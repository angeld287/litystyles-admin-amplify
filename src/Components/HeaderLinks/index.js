import React from 'react';
import currentUserContext from '../../context/currentUser/currentUser.Context';
import { LogOut } from '../Authentication/AuthComponent';
import { Menu } from 'antd';

import { redirect } from '../../context/communContex/commun.util';
import { PAGE_OPTIONS } from '../../context/communContex/commun.data';

const HeaderLinks = () => {
	const user = React.useContext(currentUserContext);

	const handleClick = (e) => {
		redirect(e.item.props.link);
	}

	const logOut = () => {
		LogOut();
		user.onUserLogOut();
	}

	const logIn = () => {
		redirect('/signin');
	}

	const rightStyle = { position: 'absolute', top: 0, right: 0 }

	const _user = user.user !== null ? user.user : JSON.parse(sessionStorage.getItem('CURRENT_USER_SESSION'))

	return (
		<>
			<Menu selectedKeys={[1]} mode="horizontal">
				{PAGE_OPTIONS.map(_ => <Menu.Item onClick={handleClick} link={_.link} key={_.id} >{_.title}</Menu.Item>)}
			</Menu>
			<Menu mode="horizontal" style={rightStyle}>
				{_user !== null && _user.signInUserSession !== null
					?
					<Menu.Item key="logoutbtn" onClick={logOut}>
						{_user.signInUserSession.idToken.payload.email} - Logout
					</Menu.Item>
					:
					<Menu.Item key="loginbtn" onClick={logIn}>
						Login
					</Menu.Item>
				}
			</Menu>
		</>
	);
}

export default HeaderLinks;
