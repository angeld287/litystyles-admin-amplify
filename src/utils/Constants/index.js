const SIZE_1024_768 = { name: "_1024_768", dimensions: { width: 1024, height: 768 } }
const SIZE_480_320 = { name: "_480_320", dimensions: { width: 480, height: 320 } }
const SIZE_320_240 = { name: "_320_240", dimensions: { width: 320, height: 240 } }

export const IMAGES = { SIZE_1024_768, SIZE_480_320, SIZE_320_240 };

