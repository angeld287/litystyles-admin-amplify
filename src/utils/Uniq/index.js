export const uniq = (a, param)=>{
    return a.filter(function(item, pos, array){
		return array.map(function(mapItem){ return mapItem[param]; }).indexOf(item[param]) === pos;
	})
}