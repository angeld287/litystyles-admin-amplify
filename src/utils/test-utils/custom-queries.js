import { queryHelpers, buildQueries } from '@testing-library/react'

// The queryAllByAttribute is a shortcut for attribute-based matchers
// You can also use document.querySelector or a combination of existing
// testing library utilities to find matching nodes for your query
const queryAllByDataCy = (...args) =>
    queryHelpers.queryAllByAttribute('data-cy', ...args)

const queryAllByAriaOwns = (...args) =>
    queryHelpers.queryAllByAttribute('aria-owns', ...args)

const queryAllByClass = (...args) =>
    queryHelpers.queryAllByAttribute('class', ...args)

const queryAllByType = (...args) =>
    queryHelpers.queryAllByAttribute('type', ...args)

const getMultipleError = (c, value) =>
    `Found multiple elements with the attribute: ${value}`
const getMissingError = (c, value) =>
    `Unable to find an element with the attribute: ${value}`

const [
    queryByDataCy,
    getAllByDataCy,
    getByDataCy,
    findAllByDataCy,
    findByDataCy,
] = buildQueries(queryAllByDataCy, getMultipleError, getMissingError)

const [
    queryByClass,
    getAllByClass,
    getByClass,
    findAllByClass,
    findByClass,
] = buildQueries(queryAllByClass, getMultipleError, getMissingError)

const [
    queryByAriaOwns,
    getAllByAriaOwns,
    getByAriaOwns,
    findAllByAriaOwns,
    findByAriaOwns,
] = buildQueries(queryAllByAriaOwns, getMultipleError, getMissingError)

const [
    queryByType,
    getAllByType,
    getByType,
    findAllByType,
    findByType,
] = buildQueries(queryAllByType, getMultipleError, getMissingError)

export {
    queryByDataCy,
    queryAllByDataCy,
    getByDataCy,
    getAllByDataCy,
    findAllByDataCy,
    findByDataCy,
    getAllByAriaOwns,
    queryByAriaOwns,
    getByAriaOwns,
    findAllByAriaOwns,
    findByAriaOwns,
    queryAllByAriaOwns,
    queryByClass,
    getAllByClass,
    getByClass,
    findAllByClass,
    findByClass,
    queryAllByClass,
    queryByType,
    getAllByType,
    getByType,
    findAllByType,
    findByType,
}