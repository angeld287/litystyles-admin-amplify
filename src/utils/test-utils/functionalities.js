import { screen, fireEvent, prettyDOM, waitFor } from '@testing-library/react';

const writeInInputFoundByPlaceHolder = (body, placeHolder, text) => {

    const input = screen.getByPlaceholderText(placeHolder);
    fireEvent.change(input, {
        target: { value: text }
    });
    expect(input.value).toBe(text);
}

const SelectItemInModalFoundByTestId = async (body, dataTestId, option) => {

    // find select by data test id
    let select = screen.getByTestId(dataTestId).querySelector('[class=ant-select-selection-search-input]');

    // Dropdown the select
    await waitFor(() => {
        fireEvent.mouseDown(select);
    });

    // Select the option
    fireEvent.click(screen.getByText(option));

    // Close the select
    fireEvent.click(body);
}

const insertFileInInputFileFoundByTestId = async (body, dataTestId, file) => {
    //Get the input file by test id
    let _file = screen.getByTestId(dataTestId);

    //verify if the input file is empty
    expect(_file.files.length).toBe(0);

    //add file to the input file
    await waitFor(() => {
        fireEvent.change(_file, {
            target: { files: [file] },
        })
    });

    //verify if the input file has the new file
    _file = screen.getByTestId(dataTestId);
    expect(_file.files.length).toBe(1);
}

export {
    writeInInputFoundByPlaceHolder,
    SelectItemInModalFoundByTestId,
    insertFileInInputFileFoundByTestId
}