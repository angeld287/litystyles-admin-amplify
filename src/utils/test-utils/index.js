// test-utils.js
import { render, queries } from '@testing-library/react'
import * as customQueries from './custom-queries'
import * as customFunctionalities from './functionalities';

const customRender = (ui, options) =>
    render(ui, { queries: { ...queries, ...customQueries, ...customFunctionalities }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }