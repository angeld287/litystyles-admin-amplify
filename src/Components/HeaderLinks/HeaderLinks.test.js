import { render, waitFor } from '../../utils/test-utils';
import HeaderLinks from './index';

test('Verify correct renderization of HeaderLinks Component', async () => {
    const { asFragment } = render(<HeaderLinks />);
    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
})