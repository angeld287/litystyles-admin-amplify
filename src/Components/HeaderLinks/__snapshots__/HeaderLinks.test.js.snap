// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify correct renderization of HeaderLinks Component 1`] = `
<DocumentFragment>
  <ul
    class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light"
    data-menu-list="true"
    role="menu"
    tabindex="0"
  >
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      data-menu-id="rc-menu-uuid-test-1"
      link="business"
      role="menuitem"
      style="opacity: 1; order: 0;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        Negocio
      </span>
    </li>
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      link="clasifications"
      role="menuitem"
      style="opacity: 1; order: 1;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        Clasificaciones
      </span>
    </li>
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      link="clients"
      role="menuitem"
      style="opacity: 1; order: 2;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        Clientes
      </span>
    </li>
    <li
      aria-hidden="true"
      class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
      role="none"
      style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
    >
      <div
        aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        data-menu-id="rc-menu-uuid-test-rc-menu-more"
        role="menuitem"
        tabindex="-1"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
  <div
    aria-hidden="true"
    style="display: none;"
  />
  <ul
    class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light"
    data-menu-list="true"
    role="menu"
    style="position: absolute; top: 0px; right: 0px;"
    tabindex="0"
  >
    <li
      class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      data-menu-id="rc-menu-uuid-test-loginbtn"
      role="menuitem"
      style="opacity: 1; order: 0;"
      tabindex="-1"
    >
      <span
        class="ant-menu-title-content"
      >
        Login
      </span>
    </li>
    <li
      aria-hidden="true"
      class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal ant-menu-submenu-disabled"
      role="none"
      style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
    >
      <div
        aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
        aria-disabled="true"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        data-menu-id="rc-menu-uuid-test-rc-menu-more"
        role="menuitem"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
  <div
    aria-hidden="true"
    style="display: none;"
  />
</DocumentFragment>
`;
